pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-headers.adoc

```diff
@@ -26,40 +33,45 @@ You can access headers in `on_request` or `on_response` wrapped functions by imp
 ----
 async fn request_filter(request_state: RequestState, _config: &Config) {
     let headers_state = request_state.into_headers_state().await;
-    let old_value = headers_state.header("request-header").unwrap_or_default();
+    let headers_handler = headers_state.handler();
+    let old_value = headers_handler.header("request-header").unwrap_or_default();
 
     let new_value = "--replaced--";
     logger::info!("Old request header value: {old_value}, New value: {new_value}");
-    headers_state.set_header("request-header", new_value);
+    headers_handler.set_header("request-header", new_value);
 }
 
 async fn response_filter(response_state: ResponseState, _config: &Config) {
     let headers_state = response_state.into_headers_state().await;
-    let old_headers = headers_state.header("request-header").unwrap_or_default();
+    let headers_handler = headers_state.handler();
+    let old_headers = headers_handler.header("request-header").unwrap_or_default();
 
     let new_value = vec![("response-header1", "--replaced--"), ("response-header2", "--replaced--")];
     logger::info!("Old request header value: {old_headers:?}, New value: {new_value:?}");
-    headers_state.set_headers(new_value);
+    headers_handler.set_headers(new_value);
 }
-
 ----
 
-Envoy handles the method, scheme, path, authority, and status codes as headers. Modify the headers by accessing the request's `:method`, `:scheme`, `:path`, `:authority`, and `:status`. PDK provides the `RequestHeadersHandler` and `ResponseHeadersHandler` traits for ease of access:
+Envoy handles the method, scheme, path, authority, and status codes as headers. Modify the headers by accessing the request's `:method`, `:scheme`, `:path`, `:authority`, and `:status`. PDK provides methods for ease of access:
 
 [source,Rust]
 ----
-pub trait RequestHeadersHandler: HeadersHandler {
-    fn method(&self) -> String;
-    fn scheme(&self) -> String;
-    fn authority(&self) -> String;
-    fn path(&self) -> String;
+async fn request_filter(request_state: RequestState, _config: &Config) {
+    let headers_state = request_state.into_headers_state().await;
+    let method = headers_state.method();
+    let scheme = headers_state.scheme();
+    let authority = headers_state.authority();
+    let path = headers_state.path();
+    ...
 }
 
-pub trait ResponseHeadersHandler: HeadersHandler {
-    fn status_code(&self) -> u32;
+async fn response_filter(response_state: ResponseState, _config: &Config) {
+    let headers_state = response_state.into_headers_state().await;
+    let status = headers_state.status_code();
 }
 ----
 
+[[read-the-request-body]]
 == Read the Request Body
 
 To access the body in both the request and the response, transform the `RequestState` or `ResponseState` to a body state by calling the method `into_body_state` and awaiting it. If the original state was already transformed into a header state, transform the state into a body state by calling the same function, for example:
```

Comment:
we could also add the parenthesis here:
`into_body_state` => `into_body_state()`